### PR TITLE
Sort entries of HEADERS, SOURCES and FORMS in *.pro files

### PR DESCRIPTION
**Pull request by ubruhin**
The new script `sort_qmake_file_entries.py` sorts the entries of HEADERS, SOURCES and FORMS variables in all qmake project files (*.pro) in alphabetical order (except those in submodules). This reduces the risk of conflicts when merging or rebasing commits which modify *.pro files and makes those files cleaner and thus easier to read.

It may even be used as a pre-commit hook.

See also https://bugreports.qt.io/browse/QTCREATORBUG-553.

### CODE REVIEW

File: dev/sort_qmake_file_entries.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import os 
+import glob
+from shutil import copyfile
```
Comment (rnestler) [inline dev/sort_qmake_file_entries.py]: Add `from __future__ import print_function` and use `print()` in the rest of the code to be python3 compatible.

File: dev/sort_qmake_file_entries.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+# search all *.pro files and sort their HEADERS, SOURCES and FORMS entries
+print "Sort *.pro file entries..."
+modified_files = sort_qmake_files_in_dir(project_root_dir, project_root_dir)
+print "Finished: {} Files modified.".format(modified_files)
+
+# exit with code != 0 if files were modified
+# -> aborts the commit when used as pre-commit hook!
+exit(modified_files)
+
```
Comment (rnestler) [inline dev/sort_qmake_file_entries.py]: It would be nicer to put everything in a `main()` function.

File: dev/sort_qmake_file_entries.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+"""
+
+
+def get_repository_root_dir(start_dir):
+    if os.path.isdir(os.path.join(start_dir, '.git')):
+        return start_dir
+    else:
+        parent_dir = os.path.dirname(start_dir)
+        if parent_dir != start_dir:
+            return get_repository_root_dir(parent_dir)
```
Comment (dbrgn) [inline dev/sort_qmake_file_entries.py]: Does this stop at `/`, or will you get infinite recursion?

Comment (rnestler) [inline dev/sort_qmake_file_entries.py]: `if parent_dir != start_dir:` will abort the recursion at `/` I guess?

Comment (dbrgn) [inline dev/sort_qmake_file_entries.py]: Ah, right.

File: dev/sort_qmake_file_entries.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function
```
Comment (ubruhin) [inline dev/sort_qmake_file_entries.py]: The script do not use any kind of division, so I think the `division` module should not change anything. `unicode_literals` may be nice, but I would say now it's too late to change :wink:

File: dev/sort_qmake_file_entries.py (hunk 3 above)
Comment (ubruhin) [inline dev/sort_qmake_file_entries.py]: Yes, as @rnestler noticed it should abort at `/`.

**Comment by ubruhin** [comment]
> A dry-run mode might also be useful for testing :)

Good idea, but as it is already merged (and especially because it's not a critical feature for end users, but only for developers) I won't add this now. In addition, the script already creates backups before modifying files, so things can be reverted if something goes wrong ;)

**Pull request merged**